### PR TITLE
Produce input for flavor manager.

### PR DESCRIPTION
**Pull request by garloff**
We don't want to duplicate information in our primary file SCS-Spec.MandatoryFlavors.yaml. Instead we generate the input in the wanted format for the OpenStack flavor-manager.

Addresses #216.

### CODE REVIEW

**Review by tibeer — APPROVED** [review]
awesome! can be merged once #229 is merged. Otherwise you'll get something like this:

```sh
$ python3 flavor-manager-input.py --help
Traceback (most recent call last):
  File "/Users/foo/Documents/SovereignCloudStack/standards/Tests/iaas/flavor-naming/flavor-manager-input.py", line 22, in <module>
    fnmck = importlib.import_module("flavor-name-check")
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/Users/foo/Documents/SovereignCloudStack/standards/Tests/iaas/flavor-naming/flavor-name-check.py", line 546, in <module>
    _bindir = os.environ('PATH').split(':')
TypeError: '_Environ' object is not callable
```

File: Tests/iaas/flavor-naming/flavor-manager-input.py
Comment (itrich) [inline Tests/iaas/flavor-naming/flavor-manager-input.py:117]:
```suggestion
    print(yaml.dump(outspec, default_flow_style=False, sort_keys=False), file=outfile)
```
<br/>
I'd propose to explicitly not sort keys so that `name` is always the first key.

Compare
```
mandatory:
- cpus: 1
  disk: 0
  name: SCS-1V:4
  ram: 4096
- cpus: 1
  disk: 10
  name: SCS-1V:4:10
  ram: 4096
- cpus: 2
  disk: 0
  name: SCS-2V:8
  ram: 8192
- cpus: 2
  disk: 20
  name: SCS-2V:8:20
  ram: 8192
- cpus: 4
  disk: 0
  name: SCS-4V:16
  ram: 16384
- cpus: 4
  disk: 50
  name: SCS-4V:16:50
  ram: 16384
- cpus: 8
  disk: 0
  name: SCS-8V:32
  ram: 32768
- cpus: 8
  disk: 100
  name: SCS-8V:32:100
  ram: 32768
- cpus: 1
  disk: 0
  name: SCS-1V:2
  ram: 2048
- cpus: 1
  disk: 5
  name: SCS-1V:2:5
  ram: 2048
- cpus: 2
  disk: 0
  name: SCS-2V:4
  ram: 4096
- cpus: 2
  disk: 10
  name: SCS-2V:4:10
  ram: 4096
- cpus: 4
  disk: 0
  name: SCS-4V:8
  ram: 8192
- cpus: 4
  disk: 20
  name: SCS-4V:8:20
  ram: 8192
- cpus: 8
  disk: 0
  name: SCS-8V:16
  ram: 16384
- cpus: 8
  disk: 50
  name: SCS-8V:16:50
  ram: 16384
- cpus: 16
  disk: 0
  name: SCS-16V:32
  ram: 32768
- cpus: 16
  disk: 100
  name: SCS-16V:32:100
  ram: 32768
- cpus: 1
  disk: 0
  name: SCS-1V:8
  ram: 8192
- cpus: 1
  disk: 20
  name: SCS-1V:8:20
  ram: 8192
- cpus: 2
  disk: 0
  name: SCS-2V:16
  ram: 16384
- cpus: 2
  disk: 50
  name: SCS-2V:16:50
  ram: 16384
- cpus: 4
  disk: 0
  name: SCS-4V:32
  ram: 32768
- cpus: 4
  disk: 100
  name: SCS-4V:32:100
  ram: 32768
- cpus: 1
  disk: 0
  name: SCS-1L:1
  ram: 1024
- cpus: 1
  disk: 5
  name: SCS-1L:1:5
  ram: 1024
reference:
- field: name
  mandatory_prefix: SCS-
- default: 'true'
  field: public
- default: 'false'
  field: disabled
- field: name
- field: cpus
- field: ram
- field: disk
```

to 

```
reference:
- field: name
  mandatory_prefix: SCS-
- field: public
  default: 'true'
- field: disabled
  default: 'false'
- field: name
- field: cpus
- field: ram
- field: disk
mandatory:
- name: SCS-1V:4
  cpus: 1
  ram: 4096
  disk: 0
- name: SCS-1V:4:10
  cpus: 1
  ram: 4096
  disk: 10
- name: SCS-2V:8
  cpus: 2
  ram: 8192
  disk: 0
- name: SCS-2V:8:20
  cpus: 2
  ram: 8192
  disk: 20
- name: SCS-4V:16
  cpus: 4
  ram: 16384
  disk: 0
- name: SCS-4V:16:50
  cpus: 4
  ram: 16384
  disk: 50
- name: SCS-8V:32
  cpus: 8
  ram: 32768
  disk: 0
- name: SCS-8V:32:100
  cpus: 8
  ram: 32768
  disk: 100
- name: SCS-1V:2
  cpus: 1
  ram: 2048
  disk: 0
- name: SCS-1V:2:5
  cpus: 1
  ram: 2048
  disk: 5
- name: SCS-2V:4
  cpus: 2
  ram: 4096
  disk: 0
- name: SCS-2V:4:10
  cpus: 2
  ram: 4096
  disk: 10
- name: SCS-4V:8
  cpus: 4
  ram: 8192
  disk: 0
- name: SCS-4V:8:20
  cpus: 4
  ram: 8192
  disk: 20
- name: SCS-8V:16
  cpus: 8
  ram: 16384
  disk: 0
- name: SCS-8V:16:50
  cpus: 8
  ram: 16384
  disk: 50
- name: SCS-16V:32
  cpus: 16
  ram: 32768
  disk: 0
- name: SCS-16V:32:100
  cpus: 16
  ram: 32768
  disk: 100
- name: SCS-1V:8
  cpus: 1
  ram: 8192
  disk: 0
- name: SCS-1V:8:20
  cpus: 1
  ram: 8192
  disk: 20
- name: SCS-2V:16
  cpus: 2
  ram: 16384
  disk: 0
- name: SCS-2V:16:50
  cpus: 2
  ram: 16384
  disk: 50
- name: SCS-4V:32
  cpus: 4
  ram: 32768
  disk: 0
- name: SCS-4V:32:100
  cpus: 4
  ram: 32768
  disk: 100
- name: SCS-1L:1
  cpus: 1
  ram: 1024
  disk: 0
- name: SCS-1L:1:5
  cpus: 1
  ram: 1024
  disk: 5
```

IMHO the latter is way nicer to read.

**Review by itrich — CHANGES_REQUESTED** [review]

File: Tests/iaas/flavor-naming/flavor-manager-input.py
Comment (garloff) [reply Tests/iaas/flavor-naming/flavor-manager-input.py:117]: Ack.

**Comment by garloff** [comment]
New feature: We add the old name into the description as alias, just in case we programmaticallly want to know ...

Please re-review, @tibeer , @itrich.

**Review by itrich — APPROVED** [review]
LGTM!

**Comment by garloff** [comment]
Sidenote 1: If we give up on mandatory flavor naming (which was considered in the IaaS team call today), I would still consider the naming standard as a recommendation, so this remains useful.
Sidenote 2: If we delay the adoption of flavor-naming-spec v2, we can always pass the option `-1` to the script to generate the old v1 names (SCS-2V:4:10) instead of the new v2 ones (SCS-2V-4-10).

**Pull request merged**
